### DISCUSSION TITLE
.NET MAUI setup instruction

### DISCUSSION THREAD

**Discussion by ajunavarghesegray** [dansiegel/xappium.uitest#75] [Q&A]
I am looking for step by step instructions for Xappium UI Test setup for .NET MAUI @dansiegel

**Comment by zokirjanov** [comment]
1. Add the Xappium.UIAutomation NuGet package: In the MyApp folder, navigate to the folder containing your UI test project (typically named MyApp.Tests). Then, add the Xappium.UIAutomation NuGet package by running the following command:


>>> dotnet add package Xappium.UIAutomation --version 1.0.0-beta1 <<<

Make sure to replace 1.0.0-beta1 with the latest version of the package.

2. Configure your tests: In your test project, create a new class that inherits from Xappium.AppHostBase. This class will serve as the base class for all your test classes. You'll need to override the GetAppConfiguration method and provide the required configuration for your app. Here's an example:


using System;
using Xappium;
using Xappium.Configuration;

public class MyAppTestHost : AppHostBase
{
    protected override AppConfiguration GetAppConfiguration()
    {
        return new AppConfiguration
        {
            App = MyApp.AppPath, // Path to your app's executable or package
            Platform = Platform.Current, // Current platform being executed
            DeviceName = "Pixel 3", // The device name (for Android or iOS)
            PlatformVersion = "10.0", // The platform version (for Android or iOS)
            AutomationName = AutomationName.Appium // The automation framework to use (Appium, in this case)
        };
    }
}

3. Create your test class: Create a new test class that inherits from your MyAppTestHost class and uses the Xappium.UIAutomation namespace. Here's an example:



using Xunit;
using Xappium.UIAutomation;

public class MyTests : MyAppTestHost
{
    [Fact]
    public void TestMyApp()
    {
        // Your test code here, for example:
        App.Tap(x => x.Marked("MyButton"));
        App.WaitForElement(x => x.Marked("MyLabel"));
        Assert.True(App.Query(x => x.Marked("MyLabel")).Any());
    }
}
4. Run your tests: To execute your tests, run the following command in the terminal or command prompt within the MyApp.Tests folder:

dotnet test
This command will build your test project, start the configured device or emulator, install your app, and run the tests.

And that's it!
